upstairs/src/lib.rs

```diff
@@ -1115,12 +1165,35 @@ async fn looper(
             }
         };
 
+        let tcp = {
+            let tls_context = tls_context.lock().await;
+            if let Some(ref tls_context) = *tls_context {
+                // XXX these unwraps are bad!
```

Comment:
Unfortunately I don't know enough to say yes or no. I was wondering how difficult it would be to check the exit status of looper and restart it - I don't think the Upstairs currently does.

Reply:
The challenge with restarting looper is it has half the message channels, and the other half live in other tasks.  So, if we exit the looper task, we have to reconnect all those channels somehow.

Reply:
If we just put a TODO: around these unwraps, then we can at least start testing with TLS and figure out if the unwraps do happen under normal operations (as opposed to buggy programs) and fix them later.  There are already a bunch of places where we need to find and handle unwraps().